packages/lib/src/FormControllerClass/FormControllerClass.ts

```diff
@@ -429,33 +435,31 @@ export class FormControllerClass {
       return;
     }
 
+    const field = this.fields.get(fieldName)!;
+
     runInAction(() => {
       this.setIsValidating(true);
-      const field = this.fields.get(fieldName)!;
       this.setFieldMeta(field, {
         isValidating: true,
       });
     });
 
-    const errors = (await (async () => {
-      try {
-        const result = await this.runFieldLevelValidation(fieldName);
-        if (result !== undefined && result !== null) {
-          return result;
-        }
-        return undefined;
-      } catch (e) {
-        return e;
-      }
-    })()) as FieldErrors;
+    const errors = await new Promise((resolve) => {
+      Promise.resolve(this.runFieldLevelValidation(fieldName))
+        .then((result: unknown) => {
+          resolve(result);
+        })
+        .catch((e: Error) => {
+          return resolve(e);
+        });
+    });
 
     runInAction(() => {
-      const field = this.fields.get(fieldName)!;
       this.setFieldMeta(field, {
         isValidating: false,
       });
 
-      this.options.fieldValueToFormValuesConverter.set(this.API.errors, fieldName, errors);
+      this.API.errors[fieldName] = errors;
```

Comment:
if errors will be empty, we will write key with empty value in API.errors? looks like a bug